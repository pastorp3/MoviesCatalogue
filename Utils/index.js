import { checkPropTypes } from 'prop-types';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import RootReducer from '../src/reducers/index';

export const findByTestAtrr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return component;
};

export const checkProps = (component, expectedProps) => {
  const propsErr = checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
  return propsErr;
}
