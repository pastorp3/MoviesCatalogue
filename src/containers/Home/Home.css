.App {
  text-align: center;
}

.movies-container {
  display: flex;
  flex-wrap: wrap;
  padding: 0 100px 0 100px;
}
