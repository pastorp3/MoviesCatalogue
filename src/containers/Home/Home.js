/* eslint-disable arrow-body-style, react/forbid-prop-types, max-len, radix */

import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import './Home.css';
import { useEffect, useState } from 'react';
import { fetchGenre, fetchMovies } from '../../actions/actions';
import Movie from '../../components/Movie/Movie';
import Navbar from '../../components/Navbar/Navbar';
import Filter from '../../components/Filter/Filter';
import Footer from '../../components/Footer/Footer';

const Home = ({
  movies,
  genres,
  getMovies,
  getGenres,
}) => {
  useEffect(async () => {
    await getMovies();
    await getGenres();
  }, []);
  const matchGenre = (ids) => {
    const values = [];
    let tmp = genres.slice();
    tmp = tmp.filter((genre) => ids.includes(genre.id));
    tmp.map((x) => values.push(x.name));
    return values;
  };
  const [filter, setFilter] = useState('All');
  const handleFilter = (e) => {
    setFilter(e.target.value);
  };
  const fetchedMovies = movies && movies;
  const filterMovies = movies && filter !== 'All' ? movies.filter((movie) => movie.genre_ids.includes(parseInt(filter))) : movies;

  return (
    <div>
      <Navbar />
      <Filter data={genres} change={handleFilter} />
      <div className="movies-container">
        { fetchedMovies && filterMovies.map((movie) => <Movie key={movie.id} poster={movie.poster_path} id={movie.id} movieGenres={genres && matchGenre(movie.genre_ids)} title={movie.original_title} year={movie.release_date} />) }
      </div>
      <Footer />
    </div>
  );
};

Home.propTypes = {
  movies: PropTypes.array.isRequired,
  genres: PropTypes.array.isRequired,
  getMovies: PropTypes.func.isRequired,
  getGenres: PropTypes.func.isRequired,
};

const mapStateProps = (state) => {
  return {
    movies: state.movies,
    genres: state.genre,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    getMovies: () => dispatch(fetchMovies()),
    getGenres: () => dispatch(fetchGenre()),
  };
};

export default connect(mapStateProps, mapDispatchToProps)(Home);

/* eslint-enable */
