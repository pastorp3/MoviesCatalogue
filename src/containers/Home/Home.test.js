import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import { BrowserRouter } from 'react-router-dom';
import thunk from 'redux-thunk';
import { findByTestAtrr, testStore } from '../../../Utils';
import RootReducer from '../../reducers';
import Home from './Home';

const setUp = (initialState = {}) => {
  const store = testStore(initialState);
  const wrapper = shallow(<Home store={store} />).childAt(0).dive();
  return wrapper;
};

describe('Home Component', () => {
  let wrapper;
  beforeEach(() => {
    const initialState = {
      movies: [{
        title: 'Test1',
        original_title: 'Test1',
        release_date: '01-01-21',
        poster_path: 'Test1',
        id: 1,
      }],
    };
    wrapper = setUp(initialState);
  });

  it('Should render without error', () => {
    const component = findByTestAtrr(wrapper, 'home-component');
    expect(component.length).toBe(1);
  });

  it('Should render according with snapshout whithout error', () => {
    const store = createStore(RootReducer, applyMiddleware(thunk));
    const tree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Home />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
