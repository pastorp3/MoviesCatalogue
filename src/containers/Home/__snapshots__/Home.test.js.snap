// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Should render according with snapshout whithout error 1`] = `
<div
  data-test="home-component"
>
  <nav
    data-test="navComponent"
  >
    <img
      alt="Logo"
      className="logo"
      data-test="logo"
      src="logo.png"
    />
    <a
      className="home-link"
      data-test="homeLink"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
  </nav>
  <div
    className="filter-container"
    data-test="Filter"
  >
    <span
      className="filter-text"
    >
      Filter by genre
    </span>
    <select
      data-test="Filter-select"
      name="genre"
      onChange={[Function]}
    >
      <option
        value="All"
      >
        All
      </option>
    </select>
  </div>
  <div
    className="movies-container"
  />
  <div
    className="footer"
    data-test="Footer"
  >
    <span>
      Copyright 2021
    </span>
  </div>
</div>
`;
